pydrive2/test/test_file.py

```diff
@@ -25,6 +28,49 @@
 )
 
 
+def auth_with_resource_key_mock(fetch_meta_data=False) -> GoogleAuth:
+    """
+    Create GoogleAuth with mocked inner httplib2.Http simulating need
+    for resourceKey header.
+    """
+    http_mock = MagicMock()
+
+    def resource_key_request(*args, **kwargs):
+        """httplib2.Http.request mock."""
+        if (
+            "X-Goog-Drive-Resource-Keys" in kwargs.get("headers", {})
+            and kwargs["headers"]["X-Goog-Drive-Resource-Keys"]
+            == "0BxphPoRgwhnodHNjS3JESnFNS1E/0-vjzOveuin3fnf4LUlfsD3A"
+        ):
+            if fetch_meta_data:
```

Comment:
can we check request args or something to detect actual URL that is requested vs using this single bool?